src/main/java/com/iexec/common/docker/client/DockerClientFactory.java

```diff
@@ -16,18 +16,71 @@
 
 package com.iexec.common.docker.client;
 
+import com.github.dockerjava.api.exception.DockerException;
+import com.github.dockerjava.api.model.AuthConfig;
+
+import java.util.HashMap;
+import java.util.Map;
+
 public abstract class DockerClientFactory {
 
-    private static DockerClientInstance defaultClient;
+    private static Map<String, DockerClientInstance> clientsMap = new HashMap<>();
 
+    /**
+     * Get an unauthenticated Docker client instance.
+     * 
+     * @return unauthenticated client
+     */
     public static synchronized DockerClientInstance getDockerClientInstance() {
-        if (defaultClient == null) {
-            defaultClient = new DockerClientInstance();
+        return createInstance("", "", "");
+    }
+
+    /**
+     * Get a Docker client instance that is authenticated against the default docker
+     * registry {@link AuthConfig#DEFAULT_SERVER_ADDRESS}
+     * ({@code https://index.docker.io/v1/}) using the provided credentials.
+     * 
+     * @param username
+     * @param password
+     * @throws DockerException if the authentication fails
+     */
+    public static synchronized DockerClientInstance getDockerClientInstance(
```

Comment:
Do we need this method? We might use explicit addresses to avoid confusions with "defaults"

Reply:
I agree we should treat the default docker registry as any other registry.